Social media profile link

Francesiyamu opened the pull request:
This is the code i wrote to improve my html and Css skill.  It is social media link that also supports responsiveness for different screen sizes.

Francesiyamu committed:
new commit

design up to the social links

Francesiyamu committed:
updated code

Here i made the code responsive so it can fit on both tablet and mobile

Kelvin-Ben commented on index.css:
```diff
@@ -1,14 +1,159 @@
+}
 
 
+main{
```
There are a couple of things I’d like to mention regarding your code:

1. Container Positioning:
  The container should be centered on the page, but currently, it is aligned to the left. To fix this, use the margin property (e.g., margin: 0 auto;) to position it correctly.

 2. Child Elements Fitting Inside the Container:
Ensure that all child elements fit well within the container. Right now, one of the list items is overflowing. To address this, try adding padding to the top and bottom of the container to properly encapsulate all its children.

3. Height and Width of the Main Element:
Avoid using fixed heights or widths for the main element. If a fixed size is required, consider using relative units (like percentages or vh/vw) for its children to prevent them from overflowing outside the container.

Kelvin-Ben commented on index.html:
```diff
@@ -9,26 +9,40 @@
-  London, United Kingdom
-  "Front-end developer and avid reader."
+  <main>
+    <article class="image">
```
Your use of the `<article>` element is valid, but consider whether `<figure>` might be more appropriate here, as it’s specifically designed for images. Also, the class name image could be more descriptive to reflect the purpose of the element (e.g., `profile-image`).

```
<figure class="image">
  <img src="assets/images/avatar-jessica.jpeg" alt="Jessica">
</figure>
```

Kelvin-Ben commented on index.html:
```diff
@@ -9,26 +9,40 @@
+
+    <article class="bio">
+      <q>Front-end developer and avid reader.</q>
+    </article>
```
Your use of `<article>` is valid, but consider whether simpler elements like` <div>, <header>, or <footer>` might be more appropriate here, as` <article>` is typically used for self-contained, reusable content. Also, ensure the `<h1>` is the only main heading on the page for accessibility. Lastly, the` <q>` element automatically adds quotation marks, so you don’t need to include them manually in the text. Keep up the good work!

```
<div class="profile-info">
  <h1>Jessica Randall</h1>
  <p>London, United Kingdom</p>
</div>

<div class="bio">
  <q>Front-end developer and avid reader.</q>
</div>
```

Kelvin-Ben commented on index.html:
```diff
@@ -9,26 +9,40 @@
+          <li><a href="#">Instagram</a></li>
+        </ul>
+      </nav>
+    </section>
```
Your use of `<nav>` and `<ul>` for the social links is excellent and semantically correct. However, consider adding a heading` (<h2>)` to the` <section>` for better semantics and accessibility. Additionally, use aria-label on the `<nav>` to describe its purpose for screen readers. If the links point to external websites, include `target="_blank"` and `rel="noopener noreferrer"` for security and usability. Great job overall!

```
<nav aria-label="Social media links">
  <ul>
    <li><a href="#">GitHub</a></li>
    <li><a href="#">Frontend Mentor</a></li>
    <li><a href="#">LinkedIn</a></li>
    <li><a href="#">Twitter</a></li>
    <li><a href="#">Instagram</a></li>
  </ul>
</nav>
```

Kelvin-Ben commented on index.css:
```diff
@@ -1,14 +1,159 @@
 
+@import url('https://fonts.googleapis.com/css2?family=Inter:ital,opsz,wght@0,14..32,100..900;1,14..32,100..900&display=swap');
```
Using `@import` to load the Inter font is correct, but for better performance, consider loading the font in the `<head>` of your HTML file using a `<link>` tag. This allows the browser to start downloading the font earlier. Additionally, include fallback fonts (e.g., `sans-serif`) in your font-family declaration to ensure text remains visible if the font fails to load. If you only need specific weights, narrow down the request to reduce file size. Great job using a modern and versatile font!

```
<head>
  <link rel="preconnect" href="https://fonts.googleapis.com">
  <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
  <link href="https://fonts.googleapis.com/css2?family=Inter:ital,opsz,wght@0,14..32,100..900;1,14..32,100..900&display=swap" rel="stylesheet">
</head>
```

Kelvin-Ben commented on index.css:
```diff
@@ -1,14 +1,159 @@
+
+ul li a:hover{
+    background-color: hsl(75, 94%, 57%);
+}
```
Your CSS is well-structured and functional, but here are a few suggestions for improvement:

1. Consider using a class for the links (e.g., .social-links a) to make the CSS more modular and reusable.

2. Remove the redundant semicolon in color: #fff;;.

3. Use consistent units for padding (e.g., rem or px) for better maintainability.

4. Add a transition property to the links for a smoother hover effect.

5. Ensure the hover color has sufficient contrast with the text color for accessibility.

```
ul {
  list-style-type: none;
  padding: 0;
  margin: 0;
}

ul li a {
  display: block;
  width: 100%;
  box-sizing: border-box;
  text-decoration: none;
  color: #fff;
  font-size: 14px;
  font-weight: 500;
  background-color: hsl(0, 0%, 20%);
  margin: 1rem 0;
  padding: 0.5rem 2rem; /* Adjusted for consistency */
  text-align: center;
  transition: background-color 0.3s ease; /* Added transition */
}

ul li a:visited {
  color: #fff; /* Ensure visited links match the default state */
}

ul li a:hover {
  background-color: hsl(75, 94%, 57%);
}
```

Great job overall! Keep up the good work!

Kelvin-Ben commented on index.css:
```diff
@@ -1,14 +1,159 @@
+    padding: 24px 24px;
+    margin: 24px auto;
+}
+}
```
Your media queries are a good start for responsive design, but here are a few suggestions for improvement:

1. Reorder the media queries so the smaller breakpoint (375px) comes before the larger one (768px) to avoid unintended overrides.

2. Move common styles like width: 100% and height: 100% outside the media queries to avoid redundancy.

3. Avoid setting max-width and max-height on the body unless necessary, as it can limit the page's responsiveness.

4. Consider using relative units like rem or % for padding and margins to make the layout more flexible.

5. Ensure the total width (width + padding + margin) does not exceed the viewport width to prevent horizontal scrolling.

```
body {
  width: 100%;
  height: 100%;
  padding: 0;
  margin: 0;
}

main {
  width: 100%;
  max-width: 327px;
  margin: 24px auto;
  padding: 24px;
}

/* Media query for screens ≥ 375px */
@media screen and (min-width: 375px) {
  main {
    max-width: 327px;
    padding: 24px;
  }
}

/* Media query for screens ≥ 768px */
@media screen and (min-width: 768px) {
  main {
    max-width: 384px;
    padding: 40px;
  }
}
```

Great job thinking about responsiveness! Keep refining your approach for even better results.

Kelvin-Ben commented on index.html:
```diff
@@ -9,26 +9,40 @@
   
```
```
<head>
  <link rel="preconnect" href="https://fonts.googleapis.com">
  <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
  <link href="https://fonts.googleapis.com/css2?family=Inter:ital,opsz,wght@0,14..32,100..900;1,14..32,100..900&display=swap" rel="stylesheet">
</head>
```

Kelvin-Ben reviewed (changes requested):
### Hello Francesiyamu,

You’ve put in a tremendous effort on this project, and it’s clear that you’re making great progress! 💯

To make this project even more amazing and ensure it aligns with the requirements, I’ve provided some suggestions for improvement. Please review the feedback carefully and implement the necessary changes.

Note:

- Use same branch to work on the requested changes.

- Once you’re done, submit a pull request so I can review your updates.

Keep up the fantastic work, and don’t hesitate to reach out if you have any questions! 🚀

Francesiyamu committed:
Corrections based on feddback

in this new update, i made adjustment to my code based on the feedback i recieved. For example i was able to align my container to the center and i figured out the problem was from my media queries. Also i removed the fixed heights and widths unless they were necessary which mostly weren't.
i also ajdusted the semantic html to simple div instead of article and for the image i placed that in a figure.  For the media queries, i re-ordered them in ascending order.

Kelvin-Ben reviewed (approved):
Hello Frances,
You have improve the codebase and am impressed by your hard work.
I do not have any further changes that are really needed.
Congrats! You can merge this brance to main 🥇
